base/shell/explorer/explorer.cpp

```diff
@@ -49,6 +49,41 @@ static VOID InitializeAtlModule(HINSTANCE hInstance, BOOL bInitialize)
     }
 }
 
+static BOOL
+GetServerAdminUIDefault()
+{
+    BOOL server;
+    DWORD value = 0, size = sizeof(value);
+    LPCWSTR rosregpath = L"SYSTEM\\CurrentControlSet\\Control\\ReactOS\\Settings\\Version";
+    if (SHGetValueW(HKEY_LOCAL_MACHINE, rosregpath, L"ReportAsWorkstation", NULL, &value, &size) == NO_ERROR)
+        server = value == 0;
+    else
+        server = IsOS(OS_ANYSERVER);
+    return server && IsUserAnAdmin();
+}
+
+static void
+InitializeServerAdminUI()
+{
+    HKEY hKey = SHGetShellKey(SHKEY_Root_HKCU | SHKEY_Key_Explorer, L"Advanced", TRUE);
+    if (hKey)
+    {
+        DWORD value, size = sizeof(value), type;
+        DWORD error = SHGetValueW(hKey, NULL, L"ServerAdminUI", &type, &value, &size);
+        if (error || type != REG_DWORD || size != sizeof(value))
+        {
+            value = GetServerAdminUIDefault();
+            if (value)
+            {
+                // TODO: Apply registry tweaks with RegInstallW; RegServerAdmin in the REGINST resource in shell32.
+                SystemParametersInfo(SPI_SETKEYBOARDCUES, 0, IntToPtr(TRUE), SPIF_SENDCHANGE | SPIF_UPDATEINIFILE);
```

Comment:
What does this have to do with the underlining of menu access letters?

Reply:
What is "this"? The comment?

Windows applies several registry values when OS_SERVERADMINUI is set to true. All the other values come from the .inf inside shell32 but keyboard cues was not present there so it must be setting it with the API.

Reply:
That question was worded badly, sorry.

Why does changing the 'SERVERADMINUI' broadcast a 'menu access letters are changed' message?
Is this one of the settings that depends on `OS_SERVERADMINUI` ?

Reply:
> Is this one of the settings that depends on `OS_SERVERADMINUI`?

SETKEYBOARDCUES does not depend on OS_SERVERADMINUI but when Explorer sets OS_SERVERADMINUI to true, it tweaks the system to what Microsoft believes advanced users/admins want. If you look at the REGINST shell32 resource on 2003 you see it will also turn on show hidden files and tweak what is visible in the Start menu (the RegShellServerAdmin "AddReg" section).